bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -480,9 +517,25 @@ public String toString() {
      */
     public void asyncReadEntries(long firstEntry, long lastEntry,
                                  ReadCallback cb, Object ctx) {
+
         // Little sanity check
-        if (firstEntry < 0 || lastEntry > lastAddConfirmed
-                || firstEntry > lastEntry) {
+        if (firstEntry < 0 || firstEntry > lastEntry) {
+            cb.readComplete(BKException.Code.ReadException, this, null, ctx);
+            return;
+        }
+
+        // Check  if need to send explicit readLastConfirmed
+        if (firstEntry > lastAddConfirmed || lastEntry > lastAddConfirmed) {
```

Comment:
Can you call readLastAddConfirmed() first to advance the lac and then call readEntries()? The change here basically change the semantic of readEntries. it would potentially break other applications.